src/hip_memory.cpp

```diff
@@ -1577,7 +1577,21 @@ hipError_t ihipMemPtrGetInfo(void* ptr, size_t* size) {
 }
 
 template <typename T>
-void ihipMemsetKernel(hipStream_t stream, T* ptr, T val, size_t count) {
+void ihipMemsetKernel(hipStream_t stream, T* ptr, T val, size_t count, bool isAsync) {
+    // Just Use count, instead of dividing by 4, the calling API already does it
+    if (sizeof(T) == sizeof(uint32_t) && (count % sizeof(uint32_t) == 0) && !isAsync) {
```

Comment:
`!isAsync` check at the beginning looks more preferable, otherwise LGTM.

Reply:
Fixed by https://github.com/ROCm-Developer-Tools/HIP/pull/1495/commits/26d166127649674295668c3fe9c9b455ece6f1ee .